tools/misc/split_coco.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import os.path as osp
+
+import mmcv
+import numpy as np
+
+
+def split_coco(data_root, fold, percent):
+    """Split COCO data for Semi-supervised object detection.
+
+    Args:
+        data_root: root of dataset
+        fold: the fold of dataset and set as random seed for data split
+        percent: percentage of labeled data
+    """
+
+    def save_anno(name, images, annotations):
+        print(f'Starting to split data {name}.json '
+              f'saved {len(images)} images and {len(annotations)}annotations')
+
+        sub_annos = {}
+        sub_annos['images'] = images
+        sub_annos['annotations'] = annotations
+        sub_annos['licenses'] = annos['licenses']
+        sub_annos['categories'] = annos['categories']
+        sub_annos['info'] = annos['info']
+        # the path of semi-supervised annotations is ./data/coco_semi_annos,
+        # because ./data/coco may be a soft link
+        path = '/'.join(ann_file.split('/')[:-3] + ['coco_semi_annos'])
```

Comment:
Specifying a separate "out-dir" would be better since users may not even have access to the soft link as well.